cmd/read.go

```diff
@@ -35,22 +35,29 @@ If --pcrs is not provided, all pcrs are read for that hash algorithm.`,
 		}
 		defer rwc.Close()
 
-		sel := tpm2.PCRSelection{Hash: pcrHashAlgo, PCRs: pcrs}
-		if len(sel.PCRs) == 0 {
-			sel = client.FullPcrSel(sel.Hash)
+		if pcrHashAlgo != tpm2.AlgUnknown {
+			sel := tpm2.PCRSelection{Hash: pcrHashAlgo, PCRs: pcrs}
+			if len(sel.PCRs) == 0 {
+				sel = client.FullPcrSel(sel.Hash)
+			}
+
+			fmt.Fprintf(debugOutput(), "Reading %v PCRs (%v)\n", sel.Hash, sel.PCRs)
+			pcrList, err := client.ReadPCRs(rwc, sel)
+			if err != nil {
+				return err
+			}
+			return pcrList.PrettyFormat(dataOutput())
 		}
 
-		fmt.Fprintf(debugOutput(), "Reading pcrs (%v)\n", sel.PCRs)
-		pcrList, err := client.ReadPCRs(rwc, sel)
+		fmt.Fprintln(debugOutput(), "Reading all PCRs")
+		banks, err := client.ReadAllPCRs(rwc)
```

Comment:
Here we are ignoring the user input --pcrs flag and print out all pcrs. We should print the --pcrs flag selection

Reply:
I added a check requiring `--hash-algo` to be specified if `--pcrs` is.

Reply:
Note this is a breaking change but that's fine.